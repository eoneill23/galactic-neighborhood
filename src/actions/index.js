export const addAPOD = APOD => ({
  type: 'ADD_APOD',
  APOD
});

export const addAPODError = error => ({
  type: 'APOD_ERROR',
  error
});

export const addISS = ISS => ({
  type: 'ADD_ISS',
  ISS
});

export const addISSError = error => ({
  type: 'ISS_ERROR',
  error
});

export const addAsteroids = asteroids => ({
  type: 'ADD_ASTEROIDS',
  asteroids
});

export const addAsteroidsError = error => ({
  type: 'ASTEROIDS_ERROR',
  error
});

export const addCalculation = calculation => ({
  type: 'ADD_CALCULATION',
  calculation
});

export const resetStoreCalculation = number => ({
  type: 'RESET_CALCULATION',
  number
});