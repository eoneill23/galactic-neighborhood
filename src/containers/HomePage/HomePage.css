.home-container {
  background-color: black;
  display: grid;
  grid-column-gap: 20px;
  grid-template-areas:
    'img' 'asteroids' 'iss'
    'img' 'asteroids' 'iss';
  grid-template-columns: 58% 18% 18%;
  height: 520px;
  padding: 30px;
}

.img-header {
  margin-bottom: 10px;
}

.img-container {
  grid-area: 'img'
}

h2 {
  color: white;
  margin: 0;
}

.asteroids-list {
  color: white;
  grid-area: 'asteroids';
  height: 430px;
  overflow: scroll;
  padding: 0;
}

.iss-container {
  grid-area: 'iss';
}

.apod {
  height: 400px;
  width: 100%;
}

.img-caption {
  text-align: center;
}

.img-copyright {
  font-size: .6em;
  text-align: center;
}

.map-container {
  height: 70px;
  width: 70px;  
}

.map-label {
  color: white;
  font-weight: bold;
  text-decoration: underline;
}