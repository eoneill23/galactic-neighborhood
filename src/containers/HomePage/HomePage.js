import React, { Component } from 'react';
import { fetchAPOD } from '../../util/apiCalls';
import './HomePage.css'

class HomePage extends Component {
  constructor() {
    super();
    this.state = {
      APOD: null,
      error: ''
    }
  }

  async componentDidMount() {
    const APOD = await fetchAPOD()
    if (APOD.error) {
      this.setState({error: APOD.error})
    } else {
      this.setState({APOD: APOD})
    }
  }


  render() {
    return (
      <>
      {this.state.APOD && <img src={this.state.APOD.url} />}
      </>
    )
  }
}

export default HomePage;