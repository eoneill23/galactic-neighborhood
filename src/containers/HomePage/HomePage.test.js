import React from 'react';
import { shallow } from 'enzyme';
import { HomePage, mapStateToProps, mapDispatchToProps } from './HomePage';
import { addAPOD, addAPODError, addISS, addISSError, addAsteroids, addAsteroidsError } from '../../actions/index';

describe('HomePage', () => {
  let wrapper, mockAPOD, mockAPODError, mockIss, mockIssError, mockAsteroids, mockAsteroidsError;

  beforeEach(() => {
    mockAPOD = {title: 'APOD', url: 'google.com'}
    mockAPODError = ''
    mockIss = {
      name: 'iss', 
      lat: 1, 
      lat: 2, 
      velocity: 3000,
      altitude: 400
    }
    mockIssError = ''
    mockAsteroids = [
      {name: 'Asteroid 1', 
      estimated_diameter: {
        miles: {
          estimated_diameter_min: 2,
          estimated_diameter_max: 3
        }
      },
      isHazardous: false},
      {
        name: 'Asteroid 1',
        estimated_diameter: {
          miles: {
            estimated_diameter_min: 1,
            estimated_diameter_max: 2
          }
        },
        isHazardous: true
      }
    ]
    mockAsteroidsError = ''
    wrapper = shallow(
      <HomePage 
        APOD={mockAPOD}
        mockAPODError={mockAPODError}
        iss={mockIss}
        issError={mockIssError}
        asteroids={mockAsteroids}
        asteroidsError={mockAsteroidsError}
      />
    )
  });

  it('should match the snapshot with all data passed in correctly', () => {

    expect(wrapper).toMatchSnapshot();
  });
});