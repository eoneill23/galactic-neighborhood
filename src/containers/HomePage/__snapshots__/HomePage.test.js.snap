// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should match the snapshot with all data passed in correctly 1`] = `
<section
  className="home-container"
>
  <div
    className="img-container"
  >
    <h2
      className="img-header"
    >
      Today's astronomy picture of the day:
    </h2>
    <img
      alt="The astronomy picture of the day"
      className="apod"
      src="google.com"
    />
    <p
      className="img-caption"
    >
      APOD
      .
    </p>
    <p
      className="img-copyright"
    >
      Â© 
       
    </p>
  </div>
  <section>
    <h2>
      Asteroids near Earth today:
    </h2>
    <ul
      className="asteroids-list"
    >
      <Asteroid
        key="Asteroid 1"
        maxDiam="3.000"
        minDiam="2.000"
        name="Asteroid 1"
      />
      <Asteroid
        key="Asteroid 1"
        maxDiam="2.000"
        minDiam="1.000"
        name="Asteroid 1"
      />
    </ul>
  </section>
  <div
    className="iss-container"
  >
    <h2>
      International Space Station Information
    </h2>
    <SpaceStationInfo
      altitude="400.00"
      velo="3000.00"
    />
    <h4
      className="map-label"
    >
      ISS's current location:
    </h4>
    <Wrapper />
  </div>
</section>
`;
